### PR TITLE
OHIF v3 - Support for OpenID Connect

### PR DESCRIPTION
**Pull request by swederik**
### PR Checklist

- [ ] Brief description of changes
- [ ] Links to any relevant issues
- [ ] Required status checks are passing
- [ ] User cases if changes impact the user's experience
- [ ] `@mention` a maintainer to request a review

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: extensions/cornerstone/src/initWADOImageLoader.js [hunk 1]
```diff
@@ -4,15 +4,41 @@ import dicomParser from 'dicom-parser';
...
+    },
+  };
+
+  if (!initialized) {
+    cornerstoneWADOImageLoader.webWorkerManager.initialize(config);
+    initialized = true;
+  }
+}
+
+export default function initWADOImageLoader(UserAuthenticationService) {
```
Comment (sedghi) [inline extensions/cornerstone/src/initWADOImageLoader.js:28]: Maybe we expose webworker configs here as well? We discussed `convertFloat32ToInt`

File: extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/getStudiesForPatientByStudyInstanceUID.js [hunk 2]
```diff
@@ -16,7 +18,7 @@ async function getStudiesForPatientByStudyInstanceUID(
   // Naturalized DICOM JSON when returning. It makes things like this awkward (mrn)
   if (getStudyResult && getStudyResult.length && getStudyResult[0].mrn) {
     return dataSource.query.studies.search({
-      patientId: getStudyResult[0].mrn,
+      patientId: getStudyResult[0]['00100020'],
```
Comment (sedghi) [inline extensions/measurement-tracking/src/panels/PanelStudyBrowserTracking/getStudiesForPatientByStudyInstanceUID.js:21]: The check is on `.mrn`, but we are getting it from `00100020`

File: platform/core/src/extensions/ExtensionManager.js [hunk 3]
```diff
@@ -277,13 +277,16 @@ export default class ExtensionManager {
   };
 
   _initDataSourcesModule(extensionModule, extensionId, dataSources = []) {
+    const { UserAuthenticationService } = this._servicesManager.services;
```
Comment (sedghi) [inline platform/core/src/extensions/ExtensionManager.js:280]: minor comment: does userAuthenticationService needs reset or clear upon exit as we do for these
https://bit.ly/3p9zAgt

File: platform/viewer/src/utils/initWebWorkers.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import cornerstoneWADOImageLoader from 'cornerstone-wado-image-loader';
+
+let initialized = false;
+
+export default function initWebWorkers() {
+  const config = {
```
Comment (sedghi) [inline platform/viewer/src/utils/initWebWorkers.js:6]: I might be missing something, but why do we have 3 separate `initiwebWorkers` for CornerstoneWADOImageLoader (two of them are new)

- here
- extensions/cornerstone/src/init.js
- platform/viewer/src/config.js

**Review by sedghi — COMMENTED** [review]
Minor comments, looks great otherwise

File: platform/core/src/extensions/ExtensionManager.js (hunk 3 above)
Comment (swederik) [reply platform/core/src/extensions/ExtensionManager.js:280]: I tried this but then navigating between pages (i.e. viewer and worklist) requires the user to log in again.

**Comment by igoroctaviano** [comment]
@swederik @sedghi I think I'm too late but just want to point out that I have used https://github.com/bjerkio/oidc-react (which is based on oidc-client.js that we used in OHIF) for the [SLIM](https://github.com/MGHComputationalPathology/slim) project which simplified/encapsulated a lot of OIDC related codebase.

File: extensions/cornerstone/src/initWADOImageLoader.js (hunk 1 above)
Comment (swederik) [reply extensions/cornerstone/src/initWADOImageLoader.js:26]: I'll do this in another PR

**Comment by swederik** [comment]
Thanks @igoroctaviano - I just removed https://github.com/maxmantz/redux-oidc because we don't use redux for anything else and it wasn't too hard to ditch the dependency. The handful of components for OIDC don't seem like they will be too difficult to maintain. The reason I didn't want to add another library is because this way we can implement login flows other than OpenID Connect using the same UserAuthenticationService (e.g. when people are embedding the app and have their own auth system). That might not matter as much for SLIM though.

**Pull request merged**